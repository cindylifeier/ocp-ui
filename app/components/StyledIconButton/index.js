/**
 *
 * StyledIconButton
 *
 */
import styled from 'styled-components';
import IconButton from 'material-ui/IconButton';

const StyledIconButton = styled(IconButton).attrs({
  style: { position: 'relative' },
  iconStyle: {
    width: '100%',
    height: 26,
    position: 'absolute',
    top: '0',
    right: '0',
  },
})`
  & svg {
    fill: rgb(51, 51, 51) !important;
    transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  }

  &:hover svg {
    fill: #26a69a !important;
  }
`;

StyledIconButton.propTypes = {};

export default StyledIconButton;
