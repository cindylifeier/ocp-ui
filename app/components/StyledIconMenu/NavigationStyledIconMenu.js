import React from 'react';
import NavigationMenu from '@material-ui/icons/MoreHoriz';

import StyledIconMenu from './index';
import StyledIconButton from '../StyledIconButton';

const NavigationStyledIconMenu = StyledIconMenu.extend.attrs({
  iconButtonElement: (<StyledIconButton><NavigationMenu /></StyledIconButton>),
})('');

NavigationStyledIconMenu.propTypes = {};

export default NavigationStyledIconMenu;
