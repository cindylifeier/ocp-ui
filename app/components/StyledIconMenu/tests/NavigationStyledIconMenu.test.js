import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import NavigationMenu from 'material-ui/svg-icons/navigation/menu';
import MenuItem from 'material-ui/MenuItem';

import StyledIconButton from 'components/StyledIconButton';
import NavigationStyledIconMenu from '../NavigationStyledIconMenu';

configure({ adapter: new Adapter() });

describe('<NavigationStyledIconMenu />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Arrange
      const primaryText = 'primaryText';

      // Act
      const renderedComponent = shallow((
        <NavigationStyledIconMenu>
          <MenuItem primaryText={primaryText} />
        </NavigationStyledIconMenu>));

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('style tests', () => {
    it('should have material-ui props for styling', () => {
      // Arrange
      const expectedProps = {
        anchorOrigin: { horizontal: 'right', vertical: 'top' },
        targetOrigin: { horizontal: 'right', vertical: 'top' },
        iconButtonElement: (<StyledIconButton><NavigationMenu /></StyledIconButton>),
      };
      const primaryText = 'primaryText';

      // Act
      const renderedComponent = shallow((
        <NavigationStyledIconMenu>
          <MenuItem primaryText={primaryText} />
        </NavigationStyledIconMenu>));

      // Assert
      expect(renderedComponent.props().anchorOrigin).toEqual(expectedProps.anchorOrigin);
      expect(renderedComponent.props().targetOrigin).toEqual(expectedProps.targetOrigin);
      expect(renderedComponent.props().iconButtonElement).toEqual(expectedProps.iconButtonElement);
    });
  });
});
