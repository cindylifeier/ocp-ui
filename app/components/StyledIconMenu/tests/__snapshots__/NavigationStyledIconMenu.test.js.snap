// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationStyledIconMenu /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Styled(IconMenu)>
    <StyledMenuItem__StyledMenuItem
      primaryText="primaryText"
    />
  </Styled(IconMenu)>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "anchorOrigin": Object {
        "horizontal": "right",
        "vertical": "top",
      },
      "animated": true,
      "children": <StyledMenuItem__StyledMenuItem
        primaryText="primaryText"
      />,
      "className": "sc-bwzfXH uTxCW",
      "iconButtonElement": <StyledIconButton__StyledIconButton>
        <NavigationMenu />
      </StyledIconButton__StyledIconButton>,
      "multiple": false,
      "onClick": [Function],
      "onItemTouchTap": [Function],
      "onKeyboardFocus": [Function],
      "onMouseDown": [Function],
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
      "onMouseUp": [Function],
      "onRequestChange": [Function],
      "open": null,
      "targetOrigin": Object {
        "horizontal": "right",
        "vertical": "top",
      },
      "touchTapCloseDelay": 200,
      "useLayerForClickAway": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "primaryText": "primaryText",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "anchorOrigin": Object {
          "horizontal": "right",
          "vertical": "top",
        },
        "animated": true,
        "children": <StyledMenuItem__StyledMenuItem
          primaryText="primaryText"
        />,
        "className": "sc-bwzfXH uTxCW",
        "iconButtonElement": <StyledIconButton__StyledIconButton>
          <NavigationMenu />
        </StyledIconButton__StyledIconButton>,
        "multiple": false,
        "onClick": [Function],
        "onItemTouchTap": [Function],
        "onKeyboardFocus": [Function],
        "onMouseDown": [Function],
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
        "onMouseUp": [Function],
        "onRequestChange": [Function],
        "open": null,
        "targetOrigin": Object {
          "horizontal": "right",
          "vertical": "top",
        },
        "touchTapCloseDelay": 200,
        "useLayerForClickAway": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "primaryText": "primaryText",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
